Support for Multiple Versions of Analytics.js (#7005)
segmentio/segment-docs: src/connections/sources/catalog/libraries/website/javascript/troubleshooting.md

Solution: [Follow the Analytics.js Quickstart Guide](/docs/connections/sources/catalog/libraries/website/javascript/quickstart/)

> ## Loading Multiple Instances of Analytics.js

Reviewer: We try to keep all h2's and below in sentence case!

```suggestion
## Loading multiple instances of Analytics.js
```

> ### Analytics.js Snippet Loaded More Than Once

Reviewer: ```suggestion
### Analytics.js snippet loaded more than once
```

> It is not possible to load the Analytics.js snippet twice on the same page, even if different write keys are used. Doing so often results in errors like `Uncaught RangeError: Maximum call stack size exceeded`. However, you can conditionally set the write key based on an environment variable:

Reviewer: ```suggestion
You cannot load the Analytics.js snippet twice on the same page, even if different write keys are used. Doing so might result in errors like `Uncaught RangeError: Maximum call stack size exceeded`. 

However, you can conditionally set the write key based on an environment variable:
```

> It is not possible to load the Analytics.js snippet twice on the same page, even if different write keys are used. Doing so often results in errors like `Uncaught RangeError: Maximum call stack size exceeded`. However, you can conditionally set the write key based on an environment variable:

Reviewer 2: This should be possible with the new snippet --just change one snippet to use the "segment" variable

Author: Discussing this on Slack: https://twilio.slack.com/archives/C9Z93GW76/p1725549203131189?thread_ts=1725380202.861859&cid=C9Z93GW76

Example:
```js
var writeKey;
ENV === 'production' ? writeKey = 'A' : writeKey = 'B';
```

### Multiple Versions of Analytics.js

> It is possible to load multiple versions of Analytics.js, such as a snippet version and an npm version (with different writekeys), in the same environment without them interfering with each other. This allows the npm library and the browser snippet to coexist without conflicting.Just keep in mind the following limitations with this approach:

Reviewer: ```suggestion
You can load multiple versions of Analytics.js in the same environment. For example, you could have both a snippet version and an npm version on one page, each with different write keys. This allows the npm library and the browser snippet to coexist without conflicting. 

Keep the following limitations in mind:
```

> **1. Device-Mode Destination Conflicts**: If you are using the same device-mode destination in both instances (i.e., across different writekeys), conflicts may occur. This is due to third-party scripts that don't support global instances. To avoid issues, ensure you are not using the same device-mode destination between different write keys.
> 
> **2. CDN URL Customization:** We currently do not support overriding the CDNURL when using multiple instances of Analytics.js.

Reviewer: Given that these items do not have a specific order (for example, you don't need to do one and then the other,) I think an unordered list would be ideal here!

```suggestion
- **Device-Mode Destination Conflicts**: If you are using the same device-mode destination in both instances (for example, across different writekeys), conflicts may occur. This is due to third-party scripts that don't support global instances. To avoid issues, ensure you are not using the same device-mode destination with different write keys.
- **CDN URL Customization:** Segment does not support overriding the CDNURL when using multiple instances of Analytics.js.
```
